### PR TITLE
Fix issue with start block parameters

### LINKED ISSUE #87: Start block parameter skipped when a watcher is started first time
Even if the parameters `HOME_START_BLOCK` and `FOREIGN_START_BLOCK` are configured in `.env` file they will not be used when the bridge is run first time since they are strings where as the code expects them to be integers:

https://github.com/poanetwork/bridge-nodejs/blob/8359556d9e9976a9388720d97de70e1bd7b35af8/src/watcher.js#L27

https://github.com/poanetwork/bridge-nodejs/blob/8359556d9e9976a9388720d97de70e1bd7b35af8/src/watcher.js#L106-L112

Scenario to reproduce:
1. Deploy the bridge contracts in the `erc-to-erc` mode. Store `bridgeDeploymentResults.json` created after the deployment.
2. Wait for few blocks produced
3. Transfer tokens from `Foreign` network
4. Wait for few blocks produced
5. Configure `.env` of the bridge instance and points `HOME_START_BLOCK` and `FOREIGN_START_BLOCK` with values inserted in `bridgeDeploymentResults.json` after the contracts deployment.
6. Run the bridge processes with empty Redis DB.
As soon as the `watcher-erc-affirmation-request` process started following appeared in the terminal:
```
"Found 0 Transfer events"
```
and the corresponding requests to relay assets (`transfer` calls) does not handled by the bridge.

### PR DESCRIPTION
**Pull request by fvictorio**
Closes #87.

Note that the names `HOME_START_BLOCK` and `FOREIGN_START_BLOCK` may be misleading: they are being interpreted as "last processed", so it will actually start from `HOME_START_BLOCK + 1`.

@akolotov do you think we should fix this? We can change the variable names to better reflect what they do, or we can fix it by subtracting 1 to them when they are used as last processed block numbers.

### CODE REVIEW

**Comment by fvictorio** [comment]
Done. To simplify things, I made sure that `lastProcessedBlock` is always a BN. Otherwise we are always guessing if it is a string, a number or a bignumber.

**Comment by akolotov** [comment]
You need to merge to `develop` (please create it) not to `master`.

File: src/watcher.js [hunk 1]
```diff
@@ -60,12 +64,12 @@ async function runMain({ sendToQueue }) {
...
 async function getLastProcessedBlock() {
   const result = await redis.get(lastBlockRedisKey)
-  lastProcessedBlock = result ? Number(result) : lastProcessedBlock
+  lastProcessedBlock = result ? toBN(result) : lastProcessedBlock
 }
 
 function updateLastProcessedBlock(lastBlockNumber) {
   lastProcessedBlock = lastBlockNumber
-  return redis.set(lastBlockRedisKey, lastProcessedBlock)
+  return redis.set(lastBlockRedisKey, lastProcessedBlock.toString())
```
Comment (akolotov) [inline src/watcher.js]: if I call `reset-lastBlock.sh` in CLI which type will be used to write the value to the database?

Comment (fvictorio) [reply src/watcher.js]: `reset-lastBlock.sh` updates the last block processed in redis. So callingit with `11` is the same as setting `START_BLOCK` to 10 and clearing the DB. Does that make sense?

**Comment by fvictorio** [comment]
> You need to merge to `develop` (please create it) not to `master`.

Oh, sorry. I just opened #103.

File: src/watcher.js (hunk 1 above)
Comment (akolotov) [reply src/watcher.js]: I meant which type of the value will be stored there: number of string?

Comment (fvictorio) [reply src/watcher.js]: Oh, sorry. All values in redis are strings as far as I know. When we obtain the value from redis, we convert it to BN.

Comment (akolotov) [reply src/watcher.js]: ok. thanks

**Pull request closed without merging**